### PR TITLE
Api 7853 retry

### PR DESCRIPTION
**Pull request by bkubiak**

### CODE REVIEW

File: agent/api_test.go [hunk 1]
```diff
@@ -1487,3 +1515,63 @@ func TestInvalidAuthorizationScheme(t *testing.T) {
...
+			retries++
+			return true
+		}
+
+		return false
+	})
+
+	err = api.Call("", nil, nil)
+	fmt.Println(retries)
+	fmt.Println(err)
```
Comment (kogierman) [inline agent/api_test.go:1539]: leftover debug prints? :D

File: internal/web_api.go [hunk 2]
```diff
@@ -145,28 +150,59 @@ func (a *fileUploadAPI) UploadFile(filename string, file []byte) (string, error)
...
+			if a.retryStrategy == nil || !a.retryStrategy(attempts, apiErr) {
+				return apiErr
+			}
+
+			token, err := a.getToken()
+			if err != nil {
+				return err
+			}
+
+			req.Header.Set("Authorization", fmt.Sprintf("Bearer %s", token.AccessToken))
```
Comment (kogierman) [inline internal/web_api.go:181]: please use `Type` from `token`, we support also `Basic` :D

File: internal/web_api.go [hunk 3]
```diff
@@ -16,13 +16,16 @@ import (
 
 const apiVersion = "3.2"
 
+type RetryStrategyFunc func(uint, error) bool
```
Comment (kogierman) [inline internal/web_api.go:19]: pls add docs for this, especially what uint is and if its enumerated from 0 or 1 :)

**Review by kogierman — CHANGES_REQUESTED** [review]

File: agent/api_test.go (hunk 1 above)
Comment (bkubiak) [reply agent/api_test.go:1539]: hahahhaha

File: internal/web_api.go (hunk 2 above)
Comment (bkubiak) [reply internal/web_api.go:181]: good catch, i fixed also `UploadFile`

File: internal/web_api.go (hunk 3 above)
Comment (bkubiak) [reply internal/web_api.go:19]: :pepe_grand:

**Review by kogierman — APPROVED** [review]

**Pull request merged**